duckduckgo/BrowserServicesKit
# Opt-outable private API usage

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1200194497630846/1204125157999066/f
What kind of version bump will this require?: Minor

**Optional**:

CC: @ayoy 

**Description**:
- No feature changes
- Initial iOS adaptation effort to make private API usage opt-outable using compilation flags



**Steps to test this PR**:
1. Disable _IS_REDIRECT_ENABLED, _MAIN_FRAME_NAVIGATION_ENABLED, _FRAME_HANDLE_ENABLED compilation flags, ensure tests pass either with those flags or without, all together or one by one
1. Disable other Navigation compilation flags, ensure build is not failing



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
